src/metabase/models/setting.clj

```diff
@@ -1079,6 +1079,26 @@
                        (tru "Using value of env var {0}" (str \$ (env-var-name setting)))
                        default)}))
 
+(defn current-user-readable-visibilities
+  "Returns a set of setting visibilities that the current user has read access to."
+  []
+  (set (concat [:public]
+               (when @api/*current-user*
+                 [:authenticated])
+               (when (has-advanced-setting-access?)
+                 [:settings-manager])
+               (when api/*is-superuser?*
+                 [:admin]))))
+
+(defn current-user-writable-visibilities
+  "Returns a set of setting visibilities that the current user has write access to."
+  []
+  (set (concat []
+               (when (has-advanced-setting-access?)
+                 [:settings-manager :authenticated :public])
+               (when api/*is-superuser?*
+                 [:admin]))))
+
 (defn admin-writable-settings
```

Comment:
Maybe this should be renamed to just `writable-settings` (or something similar) since setting managers aren't actually full admins, but can still call it.

Reply:
yes, `writable-settings` makes more sense. Renamed